File: 0921-minimum-add-to-make-parentheses-valid/0921-minimum-add-to-make-parentheses-valid.java
class Solution {
    public int minAddToMakeValid(String s) {
        int count=0;
        int add=0;
        for(char c:s.toCharArray()){
            if(c=='('){
                count++;
            }
            else{
                if(count>0)
                count--;
                else
                add++;
            }
        }
        return add+count;
    }
}

File: 0921-minimum-add-to-make-parentheses-valid/0921-minimum-add-to-make-parentheses-valid.py
class Solution(object):
    def minAddToMakeValid(self, s):
        """
        :type s: str
        :rtype: int
        """
        count=0
        add=0
        for i in s:
            if i=='(':
                count+=1
            else:
                if count>0:
                    count-=1
                else: 
                    add+=1
        return count+add